### PR TITLE
Restricting tracked/* labels to enhancements team and lead-opted-in to SIG leads

### PR DESCRIPTION
**Pull request by marosset**
Signed-off-by: Mark Rossetti <marosset@microsoft.com>

Related to https://github.com/kubernetes/sig-release/issues/2009

Discussions in https://kubernetes.slack.com/archives/C02BY55KV7E/p1662054929812849

### CODE REVIEW

File: config/prow/plugins.yaml [hunk 1]
```diff
@@ -157,6 +157,14 @@ label:
...
         # This label is used to filter/tag CVEs announced by SRC
         label: official-cve-feed
+    kubernetes/enhancements:
+      # Restrict setting of tracked/no and tracked/yes labels to the enhancements team.
+      - allowed_teams:
+        - enhancements
+        label: tracked/no
+      - allowed_teams:
+        - enhancements
+        label: tracked/yes
```
Comment (palnabarun) [inline config/prow/plugins.yaml:167]: We should also add the `tracked/out-of-tree` label.

```suggestion
        label: tracked/yes
      - allowed_teams:
          - enhancements
          label: tracked/out-of-tree
```

**Review by palnabarun — CHANGES_REQUESTED** [review]

File: config/prow/plugins.yaml (hunk 1 above)
Comment (marosset) [reply config/prow/plugins.yaml:167]: The text for `tracked/out-of-tree` says that this label is used when items do not need to be tracked by the release team

Are we sure we want to restrict this one too?
`tracked/no`: Denotes an enhancement issue is NOT actively being tracked by the Release Team
`tracked/yes`: Denotes an enhancement issue is actively being tracked by the Release Team
`tracked/out-of-tree`: Denotes an out-of-tree enhancement issue, which does not need to be tracked by the Release Team

**Comment by kikisdeliveryservice** [comment]
just adding a hold pending the call.

File: config/prow/plugins.yaml (hunk 1 above)
Comment (kikisdeliveryservice) [reply config/prow/plugins.yaml:167]: Probably yes. In the past people have incorrectly used that label and it caused issues. It makes sense for one group to own all permutations of tracked/* to ensure they are all correct.

Comment (marosset) [reply config/prow/plugins.yaml:167]: Thanks for the context!

**Pull request merged**